Sources/Support/PurchasesDiagnostics.swift

```diff
@@ -59,36 +132,107 @@ extension PurchasesDiagnostics {
         /// Failure performing a signed request
         case failedMakingSignedRequest(Swift.Error)
 
+        /// There are no offerings in project
+        case noOfferings
+
+        /// Offerings are not configured correctly
+        case offeringConfiguration([OfferingDiagnosticsPayload])
+
+        /// App bundle ID does not match the one set in the dashboard
+        case invalidBundleId(InvalidBundleIdErrorPayload?)
+
+        /// One or more products are not configured correctly
+        case invalidProducts([ProductDiagnosticsPayload])
+
+        /// The person is not authorized to make In-App Purchases
+        case notAuthorizedToMakePayments
+
         /// Any other not identifier error. You can check the undelying error for details.
         case unknown(Swift.Error)
 
     }
 
 }
 
-// MARK: - Implementation
-
 extension PurchasesDiagnostics {
+    /// A report that encapsulates the result of the SDK configuration health check.
+    /// Use this to programmatically inspect the SDK's health status after calling `healthReport()`.
+    public struct SDKHealthReport: Sendable {
+        /// The overall status of the SDK's health.
+        public let status: SDKHealthStatus
+        /// The RevenueCat project identifier associated with the current SDK configuration, if available.
+        public let projectId: String?
+        /// The RevenueCat app identifier associated with the current SDK configuration, if available.
+        public let appId: String?
+        /// The report for each of your app's products set up in the RevenueCat website
+        public let products: [ProductDiagnosticsPayload]
+        /// The report for each of your app's offerings set up in the RevenueCat website
+        public let offerings: [OfferingDiagnosticsPayload]
+
+        init(
+            status: SDKHealthStatus,
+            projectId: String? = nil,
+            appId: String? = nil,
+            products: [ProductDiagnosticsPayload] = [],
+            offerings: [OfferingDiagnosticsPayload] = []
+        ) {
+            self.status = status
+            self.projectId = projectId
+            self.appId = appId
+            self.products = products
+            self.offerings = offerings
+        }
+    }
 
-    /// Perform tests to ensure SDK is configured correctly.
-    /// - `Throws`: ``PurchasesDiagnostics/Error`` if any step fails
-    @objc(testSDKHealthWithCompletion:)
+    /// Status of the SDK Health report
+    public enum SDKHealthStatus: Sendable {
+        /// SDK configuration is valid but might have some non-blocking issues
+        case healthy(warnings: [PurchasesDiagnostics.Error])
+        /// SDK configuration is not valid and has issues that must be resolved
+        case unhealthy(PurchasesDiagnostics.Error)
```

Comment:
Question here. I see that we show all the `healthy` issues together (using `[PurchasesDiagnostics.Error]`) but for the unhealthy ones, we only show them one at a time. Is there a reason for this? It looks to me like we could have `unhealthy([PurchasesDiagnostics.Error])` to show them all at once.

Reply:
The reason why we only return one error, as opposed to the healthy with warnings case is that the BE just halts on the first error, so we should only highlight that one and there won't be any other errors at that time.

I wanted to leave warnings out of the `unhealthy` state, just so that people keep focus on the one error that's blocking 👍

Reply:
Ohh makes sense. Thanks for the explanation!